src/main/kotlin/org/rust/cargo/project/settings/ui/RustProjectSettingsPanel.kt

```diff
@@ -146,9 +160,35 @@ class RustProjectSettingsPanel(
     private val RsToolchainBase.rustup: Rustup? get() = rustup(cargoProjectDir)
 }
 
-private fun String.blankToNull(): String? = if (isBlank()) null else this
+private fun String.blankToNull(): String? = ifBlank { null }
 
 private fun wrapComponent(component: JComponent): JComponent =
     JPanel(BorderLayout()).apply {
         add(component, BorderLayout.NORTH)
     }
+
+/**
+ * Obtains a list of toolchains on a pool using [toolchainObtainer], then fills [toolchainComboBox] on the EDT.
+ */
+@Suppress("UnstableApiUsage")
+private fun addToolchainsAsync(
+    toolchainComboBox: RsToolchainPathChoosingComboBox,
+    toolchainObtainer: () -> List<Path>
+) {
+    ApplicationManager.getApplication().executeOnPooledThread {
+        val executor = AppUIExecutor.onUiThread(ModalityState.any())
+        executor.execute { toolchainComboBox.setBusy(true) }
```

Comment:
Looks like we can just call `toolchainComboBox.setBusy(true)` before `ApplicationManager.getApplication().executeOnPooledThread` since we always call `addToolchainsAsync` from EDT, can't we?